skan/csr.py

```diff
@@ -435,10 +433,40 @@ def path_means(self):
         means : array of float
             The average pixel value along each path in the skeleton.
         """
+
         sums = np.add.reduceat(self.paths.data, self.paths.indptr[:-1])
         lengths = np.diff(self.paths.indptr)
         return sums / lengths
 
+    def path_maximum_length(self):
+        """Compute the longest path.
+
+        Returns
+        -------
+        max_length : array of float
+            The maximum length in the skeleton.
+        """
+
+        sums = np.maximum.reduceat(self.paths.data, self.paths.indptr[:-1])
+
+        # We test on _testdata.skeleton1
+        # self.paths.data = [1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
+        # self.paths.indptr[:-1] = [0 11 14 17]
+        # If I understood correctly, np.ufunc.reduceat will slice an array according to a list and apply a function
+        # which means:
+        # 0-11 -->  [1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.] --> maximum --> 1.
+        # 11-14 --> [1. 1. 1. ]                        --> maximum --> 1.
+        # 14-17 --> [1. 1. 1. ]                        --> maximum --> 1.
+        # 17: -->   [1. 1. 1. 1.]                      --> maximum --> 1.
+        # which will end up with sums = [1. 1. 1. 1.]
```

Comment:
Yeah, so the problem is that the test data is not "interesting" enough for this problem. To make it more interesting, you should change the test data to be e.g. random numbers. Or whatever, like a pyramid, [1, 2, 1], [1, 2, 2, 1], [1, 2, 3, 2, 1]...?

Then, you can just return the maxes! We are only after the maximum value, right?

The key is that when you replace the values of the skeleton by the distance transform, rather than just 1s, getting the max will tell you the maximum width of the skeleton at each branch.

On the other hand, maybe what you want is the minimum width? Or the width at the midpoint of the branch? If you think about how a skeleton image tends to look, the junctions are almost always farther away from the edge of the object than the middle of the branches.

Sorry, I'm about to hop in bed, I hope that this makes sense, if not I will try to draw a diagram tomorrow, sometimes text is not enough. :blush:

Reply:
I see, I will try and come back again :) Thanks

Reply:
@jni perhaps you should try and clarify this (@snowformatics did say below they aren't sure how to follow your testing suggestions). This'll be the most important next step to unstick this PR